### PR TITLE
Repair and extend some benchmarks

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR does a deep review with validation logic on some benchmarks to ensure that we are fully testing everything needed to sensibly weight these extrinsics and pallets.

Changes:

* Handle panics/divide by zero errors when benchmark cli gets passed 0 values
* Return the pallet name when using `"*"` syntax

Update Benchmarks for
- [x] Collective
- [x] Democracy
- [x] Timestamp
- [x] Vesting

Will update Offenses in a separate PR

### CODE REVIEW

File: frame/collective/src/lib.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@ pub trait Trait<I: Instance=DefaultInstance>: frame_system::Trait {
 	type Origin: From<RawOrigin<Self::AccountId, I>>;
 
 	/// The outer call dispatch type.
-	type Proposal: Parameter + Dispatchable<Origin=<Self as Trait<I>>::Origin> + From<Call<Self, I>>;
+	type Proposal: Parameter + Dispatchable<Origin=<Self as Trait<I>>::Origin> + From<frame_system::Call<Self>>;
```
Comment (kianenigma) [inline frame/collective/src/lib.rs:68]: can you elaborate?

Comment (shawntabrizi) [reply frame/collective/src/lib.rs:68]: The original statement allowed us to pass extrinsics from the module as a `Proposal`. In the benchmark, I opted to use `System::remark` instead as my generic proposal, so I placed the requirement that I can convert from `frame_system::Call` to proposal. This trait should always be satisfied for all pallets afaik

File: frame/democracy/src/benchmarking.rs [hunk 2]
```diff
@@ -120,95 +133,185 @@ benchmarks! {
...
+			.ok_or("referendum doesn't exist")?;
+		let tally =  match referendum_info {
+			ReferendumInfo::Ongoing(r) => r.tally,
+			_ => return Err("referendum not ongoing"),
+		};
+		assert_eq!(tally.nays, 1000.into(), "changed vote was not recorded");
+	}
+
+	// Note we don't include proxy_vote_existing as it is a bit redundant
 	proxy_vote {
```
[the remark this answers is not in the document]
Comment (shawntabrizi) [reply frame/democracy/src/benchmarking.rs:212]: Yeah I can add it, just a little redundant per the comment above. But it is probably just more clear to have it, so I will add.

[the remark this answers is not in the document]
Comment (shawntabrizi) [reply frame/democracy/src/benchmarking.rs:212]: fixed

**Comment by shawntabrizi** [comment]
@thiolliere If things look good, would appreciate a green checkmark here

**Pull request merged**